File: test/fixtures/sample-js-project/payment-service.js
const stripe = require('stripe');

class PaymentProcessor {
    constructor(apiKey) {
        this.apiKey = apiKey;
    }

    charge(amount, customerId) {
        if (amount <= 0) throw new Error('Amount must be positive');
        return true;
    }
}

function processPayment(amount, customerId) {
    const processor = new PaymentProcessor('default-key');
    return processor.charge(amount, customerId);
}

module.exports = { PaymentProcessor, processPayment };

File: test/fixtures/sample-python-project/payment_service.py
"""Payment processing module."""
from typing import Optional


class PaymentProcessor:
    """Processes payments via external gateway."""

    def __init__(self, api_key: str):
        self._api_key = api_key

    def charge(self, amount: float, customer_id: str) -> bool:
        """Charge a customer."""
        if amount <= 0:
            raise ValueError("Amount must be positive")
        return True

    @staticmethod
    def validate_card(card_number: str) -> bool:
        """Validate a credit card number."""
        return len(card_number) == 16

    @property
    def is_configured(self) -> bool:
        """Check if the processor is configured."""
        return bool(self._api_key)


def create_processor(api_key: Optional[str] = None) -> PaymentProcessor:
    """Factory function for PaymentProcessor."""
    return PaymentProcessor(api_key or "default-key")
